dev/create-release/release-build.sh

```diff
@@ -110,16 +110,18 @@ fi
 # Depending on the version being built, certain extra profiles need to be activated, and
 # different versions of Scala are supported.
 BASE_PROFILES="-Pmesos -Pyarn"
-PUBLISH_SCALA_2_10=0
-SCALA_2_10_PROFILES="-Pscala-2.10"
-SCALA_2_11_PROFILES=
+
+# TODO: revisit for Scala 2.13
+
+PUBLISH_SCALA_2_11=1
```

Comment:
@vanzin @cloud-fan you may want to look at this. It's getting a little hairy in this script.

I recall that the goal was to use this script to create older Spark releases, so it needs logic for older versions. But looking at it, I don't think it actually creates quite the same release as older versions anyway. Is it OK to clean house here and assume only Spark 3 will be built from this script? I already deleted some really old logic here (Spark < 2.2)

Reply:
Just a question though, if you're releasing 2.3 wouldn't you use the release script as of 2.3? 

I think the script works for Spark 1.6 and 2.x, but not earlier versions, already. What about drawing the lines at major releases, so that this can simplify further?

Right now I think it's basically trying to support 2.2+, which are the non-EOL releases, which seems reasonable?

Reply:
>  if you're releasing 2.3 wouldn't you use the release script as of 2.3?

I started like that when I RM'ed 2.3.1, but the scripts were kinda broken that I had to fix a bunch of stuff in master. At that point, it didn't make sense to use the scripts in the 2.3 branch anymore, and keeping them in sync was kinda wasted effort.

> basically trying to support 2.2+, which seems reasonable?

That's my view. The release scripts in master should be usable for all non-EOL releases.